File: 1- Array - String/C#/5- Reverse Vowels of a String.cs
public class Solution
{
    public bool isVowel(char c)
    {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u'
            || c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
    }

    public string ReverseVowels(string s)
    {
        int l = 0, r = s.Length - 1;
        char[] str = s.ToCharArray();
        
        while (l < r)
        {
            
            if (isVowel(str[l]) && isVowel(str[r]))
            {
                // make swap
                char c = s[l];
                str[l] = str[r];
                str[r] = c;

                l++;
                r--;
            }
            else if (!isVowel(str[l]))
                l++;
            else if (!isVowel(str[r]))
                r--; 
        }
        return new string(str) ;
    }
}

File: 01- Array - String/5- Reverse Vowels of a String.cpp
class Solution {
public:
    bool isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ||
               c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
    }

    string reverseVowels(string s) {
        int l = 0, r = s.size() - 1;
        while (l < r) {
            if (isVowel(s[l]) && isVowel(s[r])) {
                char c = s[l];
                s[l] = s[r];
                s[r] = c;
                l++;
                r--;
            } else if (!isVowel(s[l])) {
                l++;
            } else if (!isVowel(s[r])) {
                r--;
            }
        }
        return s;
    }
};
